src/ring_theory/polynomial/hermite.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2023 Luke Mantle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luke Mantle
+-/
+
+import data.polynomial.derivative
+
+/-!
+# Hermite polynomials
+
+This file defines `hermite n`, the nth probabilist's Hermite polynomial.
+
+## Main definitions
+
+* `hermite n` : the nth probabilist's Hermite polynomial, defined recursively as a `polynomial ℤ`
+
+## References
+
+[Hermite Polynomials](https://en.wikipedia.org/wiki/Hermite_polynomials)
+
+-/
+
+noncomputable theory
+open polynomial
+
+/-- the nth probabilist's Hermite polynomial -/
+noncomputable def hermite : ℕ → polynomial ℤ
+| 0     := 1
+| (n+1) := X * (hermite n) - (hermite n).derivative
+
+@[simp] lemma hermite_succ {n : ℕ} : hermite (n+1) =  X * (hermite n) - (hermite n).derivative :=
+by rw hermite
+
+lemma hermite_eq_iter {n : ℕ} : hermite n = nat.iterate (λp, X*p - p.derivative) n 1 :=
```

Comment:
Lean has notation for `nat.iterate`, I think it's
```suggestion
lemma hermite_eq_iterate {n : ℕ} : hermite n = (λ p, X*p - p.derivative)^[n] 1 :=
```
I also changed the name; we rarely abbreviate function names unless they're really _really_ long.